### DISCUSSION TITLE
ArgoCD Helm

### DISCUSSION THREAD

**Discussion by danielstankw** [argoproj/argo-cd#19893] [Q&A]
Hi,
I am aware that ArgoCD under the hood uses `helm template` instead of `helm install` for helm chart deployment.
The issue with that approach for me is that in case ArgoCD fails, I am left with an empty namespace and no quick way to recover the deployment. 
My question is: Is there a way to save the manifests used for making the deployment resulting from executing `helm template` by the ArgoCD?

**Comment by danielstankw** [comment]
Up

**Reply by kostis-codefresh** [reply]
Is the namespace created by Argo CD? Or it exists in the Helm chart as a resource?

The proper way to recover would be to fix manifests and sync again. This wouldn't work in your case?

**Reply by danielstankw** [reply]
@kostis-codefresh 
Namespace is created beforehand

**Reply by kostis-codefresh** [reply]
So what exactly is the issue here? If you fix the problem and sync again doesn't everything work as expected?
